libminifi/test/libtest/integration/HTTPHandlers.cpp

```diff
@@ -453,18 +453,32 @@ void HeartbeatHandler::verifySupportedOperations(const rapidjson::Document& root
   REQUIRE(operations == std::set<std::string>(magic_enum::enum_names<minifi::c2::Operation>().begin(), magic_enum::enum_names<minifi::c2::Operation>().end()));
 }
 
-bool StoppingHeartbeatHandler::handlePost(CivetServer *, struct mg_connection *conn) {
-  verify(conn);
-  sendStopOperation(conn);
-  return true;
-}
-void StoppingHeartbeatHandler::sendStopOperation(struct mg_connection *conn) {
-  std::string resp = "{\"operation\" : \"heartbeat\", \"requested_operations\" : [{ \"operationid\" : 41, \"operation\" : \"stop\", \"operand\" : \"2438e3c8-015a-1000-79ca-83af40ec1991\"  }, "
-      "{ \"operationid\" : 42, \"operation\" : \"stop\", \"operand\" : \"FlowController\"  } ]}";
+void StoppingHeartbeatHandler::sendStartStopOperation(struct mg_connection *conn) {
+  std::lock_guard<std::mutex> lock(post_count_mutex_);
```

Comment:
If this is only used to protect the post_count_ member, then we should not hold the mutex while sending data over the network. Maybe the read and the increment should be separate steps in separate critical sections.